src/components/MyAccount/Settings/AccountSettingsTab.tsx

```diff
@@ -31,65 +31,25 @@ const AccountSettingsTab = ({ settingsData }: { settingsData: Patron }) => {
 
   const { onOpen: openModal, onClose: closeModal, Modal } = useModal()
 
-  const successModalProps = {
-    type: "default",
-    bodyContent: (
-      <Box className={styles.modalBody}>
-        <Text sx={{ marginLeft: "l" }}>
-          Your account settings were successfully updated.
-        </Text>
-      </Box>
-    ),
-    closeButtonLabel: "OK",
-    headingText: (
-      <Heading className={styles.modalHeading}>
-        <>
-          <Icon
-            size="large"
-            name="actionCheckCircleFilled"
-            color="ui.success.primary"
-          />
-          <Text sx={{ marginBottom: 0 }}> Update successful </Text>
-        </>
-      </Heading>
-    ),
-    onClose: closeModal,
-  }
-  const failureModalProps = {
-    type: "default",
-    bodyContent: (
-      <Box className={styles.modalBody}>
-        <Text sx={{ marginLeft: "l", marginRight: "m" }}>
-          We were unable to update your account settings. Please try again or{" "}
-          <Link href="https://www.nypl.org/get-help/contact-us">
-            contact us
-          </Link>{" "}
-          for assistance.
-        </Text>
-      </Box>
-    ),
-    closeButtonLabel: "OK",
-    headingText: (
-      <Heading className={styles.modalHeading}>
-        <>
-          <Icon size="large" name="errorFilled" color="ui.error.primary" />
-          <Text sx={{ marginBottom: 0 }}> Update failed </Text>
-        </>
-      </Heading>
-    ),
-    onClose: closeModal,
-  }
-
   const [isFormValid, setIsFormValid] = useState(false)
 
+  const editButtonRef = useRef<HTMLButtonElement>()
+  const firstInputRef = useRef<TextInputRefType>()
+
   const listElements = currentlyEditing ? (
     <AccountSettingsForm
+      firstInputRef={firstInputRef}
       patron={mostRecentPatronData}
       setIsFormValid={setIsFormValid}
     />
   ) : (
     <AccountSettingsDisplay patron={mostRecentPatronData} />
   )
+  useEffect(() => {
+    if (currentlyEditing === true) {
+      firstInputRef.current?.focus()
+    } else editButtonRef.current?.focus()
+  }, [currentlyEditing, firstInputRef])
```

Comment:
I'm not sure if you need `firstInputRef` in the dep array. Otherwise, why not also include `editButtonRef`?

Reply:
good point. i will remove